### ISSUE TITLE
Tabs are always counted as 1 characters; the `indent_size`/`tab_width` properties are ignored

### ISSUE THREAD

**Issue by mfederczuk** [pinterest/ktlint#1685]
When `indent_type` is set to `tab` in the `.editorconfig` file, then tabs are always counted as 1 character, regardless of what the [`indent_size`](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#indent_size)/[`tab_width`](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#tab_width) properties are set to.  
This throws of the `max-line-length` rule, and `ktlint` misses some violations.

## Expected Behavior

The character count of tab characters should be able to be controlled by the `indent_size`/`tab_width` properties.  
When the correct size of tab characters push a line over the configured max line length, then a violation of the `max-line-length` rule should be reported.

## Observed Behavior

Tabs always are counted as just one character, regardles of what any `.editorconfig` properties are set to.  
Even if a line would be too long, nothing is reported.  
For example, if tab size is configured as 4, then the "correct" total line length would be 82, but `ktlint` only counts 79.

<details>
<summary><code>$ ktlint --debug Test.kt</code> (nothing interesting in the output)</summary>

```log
14:38:27.774 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "baseline" id.
14:38:27.776 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "checkstyle" id.
14:38:27.776 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "json" id.
14:38:27.776 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "format" id.
14:38:27.776 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "html" id.
14:38:27.776 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "plain" id.
14:38:27.776 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "sarif" id.
14:38:27.776 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Initializing "plain" reporter with {verbose=false, color=false, color_name=DARK_GRAY, format=false}
[DEBUG] Rule with id 'standard:indent' should run after the rule with id 'experimental:function-signature'. However, the latter rule is not loaded and is allowed to be ignored. For best results, it is advised load the rule.
[DEBUG] Rules will be executed in order below (unless disabled):
           - standard:annotation, 
           - standard:annotation-spacing, 
           - standard:argument-list-wrapping, 
           - standard:chain-wrapping, 
           - standard:colon-spacing, 
           - standard:comma-spacing, 
           - standard:comment-spacing, 
           - standard:curly-spacing, 
           - standard:dot-spacing, 
           - standard:double-colon-spacing, 
           - standard:enum-entry-name-case, 
           - standard:filename, 
           - standard:final-newline, 
           - standard:import-ordering, 
           - standard:keyword-spacing, 
           - standard:modifier-order, 
           - standard:multiline-if-else, 
           - standard:no-blank-line-before-rbrace, 
           - standard:no-blank-lines-in-chained-method-calls, 
           - standard:no-consecutive-blank-lines, 
           - standard:no-empty-class-body, 
           - standard:no-empty-first-line-in-method-block, 
           - standard:no-line-break-after-else, 
           - standard:no-line-break-before-assignment, 
           - standard:no-multi-spaces, 
           - standard:no-semi, 
           - standard:no-trailing-spaces, 
           - standard:no-unit-return, 
           - standard:no-unused-imports, 
           - standard:no-wildcard-imports, 
           - standard:op-spacing, 
           - standard:package-name, 
           - standard:parameter-list-wrapping, 
           - standard:paren-spacing, 
           - standard:range-spacing, 
           - standard:spacing-around-angle-brackets, 
           - standard:spacing-between-declarations-with-annotations, 
           - standard:spacing-between-declarations-with-comments, 
           - standard:string-template, 
           - standard:unary-op-spacing, 
           - standard:wrapping, 
           - standard:indent, 
           - standard:trailing-comma-on-call-site, 
           - standard:max-line-length, 
           - standard:trailing-comma-on-declaration-site
14:38:28.352 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - 622ms / 1 file(s) / 0 error(s)
```
</details>

## Steps to Reproduce

1. Create a `.editorconfig` file with properties...
   * `indent_style` set to `tab`
   * `indent_size`/`tab_width` set to a numerical value greater than 1
   * `max_line_length` set to some value
2. Create a Kotlin file with a line where the total line length would be pushed over the configured max line length, if tab characters would be counted correctly
3. Invoke `ktlint`
4. No errors are reported

<details open>
<summary><code>.editorconfig</code></summary>

```editorconfig
root = true

[*]
indent_style = tab
indent_size = 4
#tab_width = 4
max_line_length = 20
```
</details>

<details open>
<summary><code>Test.kt</code></summary>

```kt
fun main() {
	// with a tab widt of 4 (as is set in the editorconfig file) this line is 22 characters long:
	"line is too long"
}
```
</details>

## Your Environment

* Version of ktlint used: `0.47.0`
* Operating System and version: Linux Fedora 36

**Comment by paul-dingemans** [comment]
I am doubting about the validity of this reasoning.

A tab is just a single character and as of that should be counted as 1. The max-line-length just evalutes the number of characters and not the equivalent in spaces.

On the other hand, I can also see value in your reasoning especially for deeper indentation levels. When tabs are counted as 1 position, the right margin isn't a straight line visually.

Changing the offset calculation to space-equivalent can lead to many follow-up issues as the line/column offsets in the Lint violations are also character based.  As this rule already exist for a couple of years, I am not inclined to consider this change unless it is widely supported by the community.

**Comment by paul-dingemans** [comment]
Won't fix
